src/Xrefcheck/Verify.hs

```diff
@@ -300,24 +321,29 @@ verifyReference
     retryVerification numberOfRetries resIO = do
       res@(VerifyResult ves) <- resIO
 
+      now <- getPOSIXTime <&> posixTimeToTimeSecond
+
+      let toSeconds = \case
+            Seconds s -> s
+            -- Calculate the seconds left until @Retry-After@ date
+            Date date -> utcTimeToTimeSecond date -:- now
```

Comment:
`a -:- b` underflows when `b < a` (see [here](https://hackage.haskell.org/package/o-clock-1.2.1.1/docs/Time-Timestamp.html#v:-45-:-45-))

```hs
λ> sec 3 -:- sec 2
1s
λ> sec 3 -:- sec 4
*** Exception: arithmetic underflow
```

This might happen when the clocks of the client and server are not synced up (and the retry-after timestamp is actually in the past, from the client's point of view). Or when the server means to tell the server to wait 1 second, but there's a stutter in the network and the response takes more than 1 second to arrive.

We should behave sensibly in this scenario (i.e. wait 0 seconds). Let's also write a test for this scenario.

Reply:
Yup, makes a great sense